internal/provisioner/kops_provisioner_cluster_installation.go

```diff
@@ -520,3 +517,22 @@ func (provisioner *KopsProvisioner) execCLI(cluster *model.Cluster, clusterInsta
 
 	return output, err
 }
+
+// Set env overrides that are required from installations for function correctly
+// in the cloud environment.
+// NOTE: this should be called whenever the Mattermost custom resource is created
+// or updated.
+func getMattermostEnvWithOverrides(installation *model.Installation) model.EnvVarMap {
+	mattermostEnv := installation.MattermostEnv
+	if mattermostEnv == nil {
+		mattermostEnv = map[string]model.EnvVar{}
+	}
+
+	mattermostEnv["MM_CLOUD_INSTALLATION_ID"] = model.EnvVar{Value: installation.ID}
+
+	if !installation.InternalFilestore() {
```

Comment:
minio supports the same encryption as AWS

Reply:
I tried enabling encryption with Minio and the Mattermost server started throwing errors and I couldn't connect. Is there maybe something that needs to be set on the Minio side?

Reply:
yes, you need to configure minio to accept that as well, but I don't know if the operator supports that, need to check the options

Reply:
Okay, I figured that may be the case. Either way, I propose that we merge this as is and update it later if we can add the operator option to support it.